### PR TITLE
Support displaying awards

### PR DESCRIPTION
**Pull request by DiegoMagdaleno**
Implements the feature request: https://github.com/spikecodes/libreddit/issues/24

TODO:

- [x] Parse awards
- [x] Make awards a property of posts
- [x] Display awards for comments
- [x] Display awards for posts
- [x] Clean up the code

I am keeping this pull request open to track the changes, and so the creation of the project is aware this issue is being worked on.

### CODE REVIEW

**Comment by DiegoMagdaleno** [comment]
This is still active, I just have been busy with exams

**Comment by spikecodes** [comment]
This is looking really good! One thing that I think might help would be a counter to show how many of each award was given to that post.

Libreddit with this PR:
image

Reddit:
image

**Comment by DiegoMagdaleno** [comment]
I see, yeah I could try working on that, and see what I can do, I don't know were reddit gets those numbers from tho

**Comment by curlpipe** [comment]
@DiegoMagdaleno Great work so far :)))

The award count seems to be located here in the JSON response
[data][children][0][data][all_awardings][AWARD NUMBER][count]

Here's an example response:

https://www.reddit.com/r/todayilearned/comments/nfdc0l/til_that_in_2019_brazilian_p%D0%BEl%D1%96ce_seized_a_parrot.json

**Comment by DiegoMagdaleno** [comment]
Thanks for the information :), I will try to implement this feature.

And thank you, I wish the code was a little cleaner

**Comment by curlpipe** [comment]
Hey, no worries, we're here to support you if you need :) and the code looks good so far. 

I also have a few comments on how to improve this PR, on the subreddit views and feed view, it doesn't seem to show awards. It only seems to show awards on the posts themselves. Perhaps it would be a good idea to add these? Also there are some instances where awards are not displayed, for example 

http://0.0.0.0:8080/r/LeopardsAteMyFace/comments/nfchyx/corporate_funded_far_right_conspiracy_theorist/ 

vs 

http://reddit.com/r/LeopardsAteMyFace/comments/nfchyx/corporate_funded_far_right_conspiracy_theorist/

**Comment by DiegoMagdaleno** [comment]
They are not displayed because in order to optimize it a bit, the idea was to show a maximum of 3 awards, and then have a button/option to show the rest of the awards.

If you have more than 3 awards they don't show.

I'll work on finishing this when I'm done with school :)

**Comment by curlpipe** [comment]
Is this to optimise space or time? As for time, limiting the number of awards won't have a significant effect on speed as all the award data is already parsed and loaded when the request is made as far as I can see.

**Comment by DiegoMagdaleno** [comment]
I see thats interesting:)

I will start working on this again next week :)

**Comment by DiegoMagdaleno** [comment]
@curlpipe @spikecodes Awards now Display on the homepage :), I think most of this PR is done! You guys can test it and tell me if there are any more features you guys want added. 

Screen Shot 2021-06-08 at 5 31 44 PM

Screen Shot 2021-06-08 at 5 31 37 PM

**Comment by DiegoMagdaleno** [comment]
Oh forgot to implement count, ill do it in a bit

**Comment by curlpipe** [comment]
Bloody good job mate. 

When the awards have counts, I'd say this would be good to go.

Looks great.

**Comment by spikecodes** [comment]
Wow, looking fantastic! :)

**Comment by DiegoMagdaleno** [comment]
@spikecodes @curlpipe This pull request has been ready to be merged since June, however, the main issue I have is with the CSS styling and my lack of knowledge on frontend, a little of help in that part would be appreciated :).

I don't know how to make this not overlap, or how to make it responsive. 

Sorry if it was staled, I forgot to mark it ready for review, it would be helpful, if someone could help with the CSS.
image

**Comment by mikupls** [comment]
@spikecodes did you have a chance to review this one? Is the code ready minus the CSS, or do you think more work is needed? If the code is good, I'm happy to take a look at the CSS a bit. (Sorry, still not good enough with Rust myself)

**Comment by DiegoMagdaleno** [comment]
@mikupls Not spikes but the backend is pretty much done, it just needs some decorative CSS and you can call it a day.

File: templates/post.html [hunk 1]
```diff
@@ -43,6 +43,18 @@
 				{% endif %}
 				<span class="dot">&bull;</span>
 				<span class="created" title="{{ post.created }}">{{ post.rel_time }}</span>
+				{% if !post.awards.is_empty() %}
+				<span class="dot">&bull;</span>
+				<span class="awards">
+					{% for award in post.awards.clone() %}
+					<span class="tooltip">
+						<img alt="{{ award.icon_url }}" src="{{ award.icon_url }}" class="awards_image"/>
+						<span class="tooltiptext">{{award.name}}</span>
```
Comment (spikecodes) [inline templates/post.html:52]: Aside from style, was there a reason for choosing a custom tooltip over the built-in [browser tooltip](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title)?

Comment (DiegoMagdaleno) [reply templates/post.html:52]: Not really, it is just for aesthetic purposes.

**Comment by spikecodes** [comment]
I've made some tweaks to the formatting of the code and the CSS and this PR looks ready to merge.

Thanks so much for your contribution @DiegoMagdaleno!

**Comment by DiegoMagdaleno** [comment]
No problem thank you for your amazing software!

**Pull request merged**